optimize.c

```diff
@@ -913,11 +913,11 @@ opt_peep(opt_state_t *opt_state, struct block *b)
 			break;
 
 		case BPF_JGT:
-			v = (unsigned)v > b->s.k;
+			v = v > b->s.k;
```

Comment:
No, that is doing exactly what it's supposed to do - convert the value to unsigned and compare it with the unsigned constant in the BPF instruction.  At run time, the comparisons done by the BPF interpreter are unsigned, so, to optimize it at compile time, we have to do an unsigned comparison.